File: mirea/cources/2026-2027/aii-july/02/Part-02-Stack-patterns/examples/example-02.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    string s;
    cin >> s;

    vector<char> st;
    bool ok = true;

    for (char ch : s) {
        if (ch == '(') {
            st.push_back(ch);
        } else {
            if (st.empty()) {
                ok = false;
                break;
            }
            st.pop_back();
        }
    }

    if (!st.empty()) {
        ok = false;
    }

    cout << (ok ? "correct" : "incorrect") << '\n';
    return 0;
}

File: mirea/cources/2026-2027/aii-july/02/Part-02-Stack-patterns/examples/example-02.py
s = input().strip()
stack = []
ok = True

for ch in s:
    if ch == '(':
        stack.append(ch)
    else:
        if not stack:
            ok = False
            break
        stack.pop()

if stack:
    ok = False

print("correct" if ok else "incorrect")